src/components/Users.jsx

```diff
@@ -0,0 +1,76 @@
+import { fetchUsers } from "../api";
+import { postComment } from "../api";
+
+import { useState } from "react";
+import { useEffect } from "react";
+
+export const Users = ({ article, comments, setComments }) => {
+  const [users, setUsers] = useState([]);
+  const [username, setUsername] = useState("");
+  const [commentPost, setCommentPost] = useState("");
+  const [commentInput, setCommentInput] = useState("");
+
+  useEffect(() => {
+    fetchUsers()
+      .then((data) => {
+        setUsers(data);
+      })
+      .catch((err) => {
+        console.log(err, "an error");
+      });
+  }, [commentPost, username]);
+
+  const handleUserChange = (event) => {
+    setUsername(event.target.value);
+
+  };
+
+  const handleCommentChange = (event) => {
+    setCommentInput(event.target.value);
+    setCommentPost(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+   
+
+    postComment(article.article_id, username, commentInput)
+      .then((data) => {
+
+        return setComments((currComments) => {
+          return [data.comment, ...currComments];
+        });
+      })
+      .catch((err) => {
```

Comment:
Remember to undo the optimistic render as if we reach the .catch() it means that the comment was not added to the db